### DISCUSSION TITLE
How to update Description on my repository?

### DISCUSSION THREAD

**Discussion by mikadarmawijaya00000** [community/community#54372] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

How to update Description on my repository?

**Comment by denitiawan** [comment, marked as answer]
Step by step for update the Description on repository :
- go to your repository
- click setting button on "top right"
- and then type on "description" input text
- click button "save changes"

finally your repo got new description

**Reply by mikadarmawijaya00000** [reply]
Thanks for your reply @denitiawan

**Reply by axedayz** [reply]
i doesent find the "description" input
![image](https://github.com/community/community/assets/124924510/5b3d5d5c-32f6-4fdb-8839-0f095abdd15c)

**Reply by denitiawan** [reply]
![image](https://github.com/community/community/assets/11941308/a753c2f6-289a-49bd-913a-4e7518bca766)

**Reply by joelmellon** [reply]
There are two Settings cogs "top right". 🤦

Thanks for the screenshot clarifying that it's the lower right of the two top rights, lol.

It does make sense in that it's right by the description, but still a confusing UX when it comes to following instructions online.

**Reply by jajocor** [reply]
In my case the second gear is not there. The whole "about" side window is missing.

**Reply by jajocor** [reply]
You also must have something in the repository in order to see the settings cogwheel where you can update the description. If you just created the repository and it is completely empty, you need to add a readme or some other file in order to see the settings.

**Reply by ChitKhonn** [reply]
I got it. Thanks

**Reply by nomanmalek18** [reply]
thank you

**Reply by manishpatel0** [reply]
I got it, thanks you

**Reply by ShanyMargalit** [reply]
in my case, there is no description button, what to do?
I have just a settings, but not description...
I'll be happy to get help.

**Reply by TheGiraffe3** [reply]
@ShanyMargalit are you on mobile or a desktop?

**Reply by DcorpProj** [reply]
thanks you

**Reply by Ahmed8881** [reply]
1-Go to your repository.
2-Click on the "Settings" button (top right).
3-Type in the new description in the "Description" input field.
4-Click the "Save changes" button.
5-Verify the updated description on the repository's main page.

**Reply by therobyouknow** [reply]
> 1-Go to your repository.
> 2-Click on the "Settings" button (top right).
> 3-Type in the new description in the "Description" input field.

The repository has to be initialised before the description panel is shown


> 4-Click the "Save changes" button.
> 5-Verify the updated description on the repository's main page.

If the repository is not initialised then the description does not show.

![Screenshot 2024-06-22 at 09 46 26](https://github.com/community/community/assets/1080646/66dacf44-4223-4d81-be17-fffc41d165dd)

**Reply by KalHal123** [reply]
Problem:
Idk where y'all are seeing this "Description input field" when I go to settings not under any tab not general not **anywhere** I can find an input field where I can change the description!! I am on PC. I am in the menu everyone told me to. I am looking I even did the old Contro +F returned nothing. But for the love of god **I can not click into an Input field that does not exist for me, I'm sorry** Sorry if I come over a bit aggresive but I just hate this in software and coding in general when I am told by everybody to "jUsT pReSs ThAt BuTtOn" but the button is non-existent for me. If anyone wants me to prove it I'd go as far as to provide Screenshots:
![image](https://github.com/user-attachments/assets/55aed445-2ad1-4d6e-af02-3f5e3d9cede1)
![image](https://github.com/user-attachments/assets/181b1066-1e38-4dbd-9b41-a9d235168c57)
^^^^ This continues for all tabs on the sidebar and I'm not seeing anything description related. Same goes for when you scroll down.

**Reply by KJSloan2** [reply]
@KalHal123 Try this. Go to the main page of your repo and click the gear icon I'm pointing to with the magenta arrow in this screenshot.  That should bring up a settings window where you can add your description. 
<img width="1277" alt="image" src="https://github.com/user-attachments/assets/7deec1bc-b58a-4a3f-81f2-d67be0d063e9">

**Reply by osparking** [reply]
This should be the respected answer.

**Reply by sean-abbott** [reply]
+1

**Reply by stratxgy** [reply]
> @KalHal123 Try this. Go to the main page of your repo and click the gear icon I'm pointing to with the magenta arrow in this screenshot. That should bring up a settings window where you can add your description. <img alt="image" width="1277" src="https://private-user-images.githubusercontent.com/61387934/352782375-7deec1bc-b58a-4a3f-81f2-d67be0d063e9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjkxOTk4MDcsIm5iZiI6MTcyOTE5OTUwNywicGF0aCI6Ii82MTM4NzkzNC8zNTI3ODIzNzUtN2RlZWMxYmMtYjU4YS00YTNmLTgxZjItZDY3YmUwZDA2M2U5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDE3VDIxMTE0N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJmNzNjOGNlZGIzZmE1MmY1YTNjZjBjODVjMzFlNWMxZDk0ZjkwMDQxODBmNTI5Y2MwN2VlNjhlYmNmNTUwZTImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.oaWOo3elKrmijP945nyigheV_wwr4ZgZ8AH6Vg9E3bc">

Correct answer.

**Reply by ScottMonolith** [reply]
Hm, I have the 'Settings' with the gear icon but I do not have this little gear on the far right...

On another repo I see that icon - perhaps because the new repo was initialized and has not been populated with any data yet?  Seems odd that I can't change the description until something is uploaded to the repo...

**Reply by marckruzik** [reply]
@ScottMonolith Yes, the "About" cog wheel does not appear if the repo is empty.
See https://github.com/orgs/community/discussions/54372#discussioncomment-6830666

**Reply by wecoxylot-hue** [reply]
> Step by step for update the Description on repository :
> - go to your repository
> - click setting button on "top right"
> - and then type on "description" input text
> - click button "save changes"
> 
> finally your repo got new description

**Reply by wecoxylot-hue** [reply]
~~~~> thanks you

**Reply by akostadinov** [reply]
Seems like this is not working for archived repos.

**Reply by rathod-0007** [reply]
That method works only for active repositories — if your repository is archived, it becomes read-only, so you can’t edit the description or any other settings.

To update the description:
Go to Settings → Danger Zone.
Click Unarchive this repository and confirm.
Then go back to the repo’s main page and click the ⚙️ (gear icon) next to the description to edit it.
After updating, you can re-archive the repo if you want.

**Comment by eduardolbruno** [comment]
Simple step by step if someone is in the same scenario:

- Open your repo
- Click “Settings” button
- Type on "description" input text all the info you want to show. 
- Don’t forget to "save changes"

Thats all! Also, you can change it anytime. It doesn’t affect your contributions or any progress related to that repo.

**Reply by Gadrawingz** [reply]
Great solution to the given query! Let's go!

**Reply by divStar** [reply]
Thank you - this worked perfectly. However: it only works in the regular desktop view, not in the mobile view (or I wasn't able to find it in the latter).

**Comment by 0dayhunter** [comment]
To update the description of your repository on GitHub, follow these steps:

1. **Log In:** Make sure you're logged in to your GitHub account.

2. **Navigate to Repository:**
   - From your GitHub dashboard, click on the repository you want to update the description for.

3. **Access Settings:**
   - In the repository, click on the "Settings" tab located on the right side of the repository menu.

4. **Edit Description:**
   - On the left sidebar of the settings page, you'll find a field labeled "Repository name" and below it, the "Description" field.
   - Click on the "Edit" button (pencil icon) next to the repository description.

5. **Update Description:**
   - Enter the new description for your repository in the provided text box.

6. **Save Changes:**
   - After updating the description, scroll down to the bottom of the page.
   - Click the "Save changes" button to save your updated repository description.

7. **Verify Changes:**
   - Go back to your repository's main page to see the updated description displayed under the repository name.

That's it! Your repository description should now be updated with the new information you provided. Remember that the repository description is a short summary that helps others understand the purpose or content of your repository.

**Reply by UTandor** [reply]
For some reason, it says this in the embed of my repo instead of the description:
"Contribute to username/repo development by creating an account on GitHub." instead of the description of the project

**Comment by afrinxnahar** [comment]
Thanks. Helped a lot

**Comment by rla** [comment]
The settings button for this is missing when there are no commits in the repository.

**Reply by hassanbasri** [reply]
yep,  same

**Reply by hangshan-boop** [reply]
it is new. Please don't modify

**Reply by therickfactr** [reply]
It was acknowledged above that you cannot modify the description of an empty repository. Create a simple README.md and you should be good to go.

**Comment by jaredperrin** [comment]
I don't see any way to modify the description from the area being described here. Attached is a screenshot from the "Settings" tab - it does not appear there is any way to change the description from here. What am I missing? 
![GitHubDescription](https://github.com/community/community/assets/163172979/e0ff0381-27f5-4dd9-96a0-3d98709da60c)

**Reply by afrinxnahar** [reply]
You can directly modify the description here anytime you want... No need to go to settings tab🙂

![gh1](https://github.com/community/community/assets/125039325/d32512f9-244a-4f8d-9ac1-bc62ed5dbb6b)

**Reply by keeheung** [reply]
Thanks a lot!! I don't know why my eyes couldn't find that....

**Reply by afrinxnahar** [reply]
Don't worry. It happens to all...

**Reply by fastoch** [reply]
@Afrin127329 
You're the best!

**Reply by reza-nzri** [reply]
Thanks a bunch! ❤️

**Reply by 09anushkam** [reply]
Thanks it helped!

**Reply by Modak-NeelKamal** [reply]
thank you

**Reply by retamalesj** [reply]
Well, keeheung, you don't know why, but, i created a new project and that button doesn't appears, so, it's completely not your problem, it's a problem of design of the app. Because "settings" also should give the option to change the description.

**Reply by ScottMonolith** [reply]
@doorkaz - evidently if the repo is empty you can't change the description... seems like a pretty big miss / bug to me, but what do I know.

https://github.com/orgs/community/discussions/54372#discussioncomment-10987548

**Reply by daniel-pfeiffer** [reply]
> You can directly modify the description here anytime you want... No need to go to settings tab🙂
> 
> ![gh1](https://private-user-images.githubusercontent.com/125039325/316837080-d32512f9-244a-4f8d-9ac1-bc62ed5dbb6b.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjI3MjM3NTIsIm5iZiI6MTc2MjcyMzQ1MiwicGF0aCI6Ii8xMjUwMzkzMjUvMzE2ODM3MDgwLWQzMjUxMmY5LTI0NGEtNGY4ZC05YWMxLWJjNjJlZDVkYmI2Yi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMTA5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTEwOVQyMTI0MTJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yM2YzMTMyYjJjNjU0OTc0NzYyYTk5NzcxY2RkM2JhYzBjMTM1YmYwOGEyNTIyNGQ2NDA1YmU3NjkyZDUwODU2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.3zd9pAFFOMK2dMRcmbHSt44K_xdegu5vHSNjbUhPDZc)

Seriously⁉️ There is a 2nd tiny unlabelled cog-wheel⁉️ Who designs such unintuitive UIs? Minutes of my life wasted…

**Reply by BradMyrick** [reply]
This should be the accepted answer.  the settings cog does not contain the description.

**Comment by balusena** [comment]
Response #2

**Comment by erwinkramer** [comment]
It's not possible to change the description of a repo in the mobile version of the website 👎

**Reply by TheGiraffe3** [reply]
If you scroll down to the very bottom, the settings button you’re looking for should be on the right.

**Comment by ShanyMargalit** [comment]
Hi, I'm on desktop.


[image: Mailtrack]
<https://mailtrack.io?utm_source=gmail&utm_medium=signature&utm_campaign=signaturevirality11&>
Sender
notified by
Mailtrack
<https://mailtrack.io?utm_source=gmail&utm_medium=signature&utm_campaign=signaturevirality11&>
18.06.24,
02:39:11

‫בתאריך יום ב׳, 17 ביוני 2024 ב-6:26 מאת ‪Loymdayddaud‬‏ <‪
***@***.***‬‏>:‬

**Reply by TheGiraffe3** [reply]
Check [above](https://github.com/orgs/community/discussions/54372#discussioncomment-8913904) then.

**Comment by ShanyMargalit** [comment]
HiHi, you probably didn't understand my question. I want to add a
description to a repository that doesn't currently have a description. What
you sent me, is to edit a description that already exists, so it's not
relevant to me. Can you help me how to add a description to the repository?




[image: Mailtrack]
<https://mailtrack.io?utm_source=gmail&utm_medium=signature&utm_campaign=signaturevirality11&>
Sender
notified by
Mailtrack
<https://mailtrack.io?utm_source=gmail&utm_medium=signature&utm_campaign=signaturevirality11&>
18.06.24,
12:39:48

‫בתאריך יום ג׳, 18 ביוני 2024 ב-6:39 מאת ‪Loymdayddaud‬‏ <‪
***@***.***‬‏>:‬

**Comment by t0linixer** [comment]
I created a repo without the description
But I'm going to try to add description in old repo, but I cant find "description" button
Please help me this issue

**Comment by ShanyMargalit** [comment]
That's my question, also!!
I don't know what to do!!
If somebody has a solution,
please help me.
thanks.




[image: Mailtrack]
<https://mailtrack.io?utm_source=gmail&utm_medium=signature&utm_campaign=signaturevirality11&>
Sender
notified by
Mailtrack
<https://mailtrack.io?utm_source=gmail&utm_medium=signature&utm_campaign=signaturevirality11&>
20.06.24,
11:49:08

‫בתאריך יום ה׳, 20 ביוני 2024 ב-9:51 מאת ‪DcorpProj‬‏ <‪
***@***.***‬‏>:‬

**Comment by fastoch** [comment]
On the web app: after opening your repo, you should see a gear icon at the top right

**Comment by ShanyMargalit** [comment]
I see this gear, but there is no option to add a description.
 I'll point out again, I opened a repository without adding a description
to it, and now when I want to add, it can't be done through the
repository's settings



[image: Mailtrack]
<https://mailtrack.io?utm_source=gmail&utm_medium=signature&utm_campaign=signaturevirality11&>
Sender
notified by
Mailtrack
<https://mailtrack.io?utm_source=gmail&utm_medium=signature&utm_campaign=signaturevirality11&>
20.06.24,
12:55:12

‫בתאריך יום ה׳, 20 ביוני 2024 ב-12:12 מאת ‪Fastoch‬‏ <‪
***@***.***‬‏>:‬

**Comment by ShanyMargalit** [comment]
And I have another small question,
 every time I upload a commit, git writes that unknown uploaded the commit,
 and it doesn't recognize that it's me, even though I logged in through my
personal account.
someone has a solution?



[image: Mailtrack]
<https://mailtrack.io?utm_source=gmail&utm_medium=signature&utm_campaign=signaturevirality11&>
Sender
notified by
Mailtrack
<https://mailtrack.io?utm_source=gmail&utm_medium=signature&utm_campaign=signaturevirality11&>
20.06.24,
12:58:17

‫בתאריך יום ה׳, 20 ביוני 2024 ב-12:55 מאת ‪Shani Margalit‬‏ <‪
***@***.***‬‏>:‬

**Comment by Rudra-Sankha-Sinhamahapatra** [comment]
![Screenshot 2024-06-20 160207](https://github.com/community/community/assets/101992909/5a5e49bf-17d3-4341-90a8-62825453de65)
Hii @mikadarmawijaya00000  At first go the home page of the repo
You can see the settings icon on the top right corner ,click it
after that you can edit the desription
![image](https://github.com/community/community/assets/101992909/19d78542-281d-459f-9a0b-8bfbade94645)

**Comment by ShanyMargalit** [comment]
You probably didn't understand my problem, my problem is not to edit a
description, but to add a description that doesn't exist, I go into the
gear and there is no description tab, my question is: how can you add a
description to a repository that doesn't have a description?



[image: Mailtrack]
<https://mailtrack.io?utm_source=gmail&utm_medium=signature&utm_campaign=signaturevirality11&>
Sender
notified by
Mailtrack
<https://mailtrack.io?utm_source=gmail&utm_medium=signature&utm_campaign=signaturevirality11&>
20.06.24,
13:36:50

‫בתאריך יום ה׳, 20 ביוני 2024 ב-13:34 מאת ‪Rudra Sankha Sinhamahapatra‬‏ <‪
***@***.***‬‏>:‬

**Reply by Rudra-Sankha-Sinhamahapatra** [reply]
Bro my repo dosent have a description clearly see it in the image

**Reply by fastoch** [reply]
Yes, everyone understood your problem. But as already stated, there should be a gear icon allowing you to add a description.
Your issue might be related to your Web browser. Have you tried another one?

**Comment by alpine-storm** [comment]
![image](https://github.com/community/community/assets/171306929/76902226-4f0d-4be3-9204-1501e6794137)

**Reply by ar-sayeem** [reply]
Finally the "Top Right" setting button I found, THANKS A LOT <3

**Reply by apoorvdarshan** [reply]
thanks a ton!

**Comment by ShanyMargalit** [comment]
OMG,
 thank you very much!!!
 I succeed to do it
 thanks all



[image: Mailtrack]
<https://mailtrack.io?utm_source=gmail&utm_medium=signature&utm_campaign=signaturevirality11&>
Sender
notified by
Mailtrack
<https://mailtrack.io?utm_source=gmail&utm_medium=signature&utm_campaign=signaturevirality11&>
20.06.24,
16:13:03

‫בתאריך יום ה׳, 20 ביוני 2024 ב-14:50 מאת ‪M Ahmed Butt‬‏ <‪
***@***.***‬‏>:‬

**Comment by BlumyLevintal** [comment]


**Reply by TheGiraffe3** [reply]
This is probably the wrong place.
That said, you look at the top right with the three buttons. Click it and press "Edit". Then you can type what you wanted to put in.

**Comment by aflacake** [comment]
Note: this cannot be done in repository private mode

**Reply by ShanyMargalit** [reply]
Hi Blumy!
Are you from Ashdod?,I think I know you...
Anyway, in the right side, you have gear, click there, and add your description.
![image](https://github.com/community/community/assets/164001455/ba0bf8bc-47da-41a5-ae5d-6d4a021e8714)

**Comment by ShanyMargalit** [comment]
I was also sure that this button was not found, until someone attached a
screenshot and beat me to it.[image: image.png]

‫בתאריך שבת, 27 ביולי 2024 ב-22:17 מאת ‪KalHal123‬‏ <‪
***@***.***‬‏>:‬

**Comment by aflacake** [comment]
# Change private repository to public
The way to do this is to edit, 
previously I wanted to edit the description because I forgot to set it at the beginning of creation
So i try it:
![ss1](https://github.com/user-attachments/assets/20b89ec1-ec59-4aee-b982-bc8a822d8e59)
There are no description settings here
![ss2](https://github.com/user-attachments/assets/b4bfef70-adca-4307-8152-2f5cdb0a9f1d)
You need to do, set it to the settings in the repository
![ss3](https://github.com/user-attachments/assets/3900b969-de3b-4a31-9c20-f0f0a69344b1)
**Change visibiliity** and click **"I want to make this repository public"**, click more another messages

It still doesn't appear, try adding a readme file or any file in your repository
![ss6](https://github.com/user-attachments/assets/7289c630-2dc3-4711-ae7b-132b4bc51ac8)
Well now it's here
![ss5](https://github.com/user-attachments/assets/0db8132f-2c53-49e8-a5b3-db4634ccdd26)

**Reply by marckruzik** [reply]
It has nothing to do with visibility. It is a GitHub bug related to empty repos.
I'm describing your steps with more details:
* You start with an empty repo.
* Empty repos have a special page "Quick setup" which is missing the right tab "About" with its gear icon to edit description.
* You change the visibility, which adds a "README.md" file in the repo.
* Because there is a file in the repo, the repo is no longer empty.
* The repo now displays the normal page, with the right tab "About" with its gear icon to edit description.

You can actually get the same result by doing the following (emphase on the step which replaces changing the visibility):
* You start with an empty repo.
* Empty repos have a special page "Quick setup" which is missing the right tab "About" with its gear icon to edit description.
* **You add any file, by commiting it or using "/new/main" ( https://github.com/YourName/YourRepo/new/main )** (there are links in the Quick setup page to create them fast)
* Because there is a file in the repo, the repo is no longer empty.
* The repo now displays the normal page, with the right tab "About" with its gear icon to edit description.

I already reported this GitHub bug, but it's still there. This is confusing a lot of people.

My issue on this GitHub bug is here: https://github.com/orgs/community/discussions/43623

**Comment by tgrushka** [comment]
This is a major accessibility issue. I am legally blind and could not find "Description" in the settings anywhere. I use search on the page to find out how to do things. The gear icon is not accessible or helpful. This is a massive oversight!! How can one "find" a gear icon on the page? Not possible. It's totally visual, and not intuitive, even for a sighted person. One would expect to go into "Settings" to edit anything related to the repo. The tiny gear icon nearly hidden away is neither intuitive, nor helpful.

**Comment by jihanurrahman33** [comment]
Hi @mikadarmawijaya00000,

You can update the description of your repository in GitHub by following these steps:

### Method 1: From the repository homepage

1. Open your repository on GitHub.
2. At the very top of the page, just under the repo name, you’ll see a **Description box** (it may say “Add a description” if you don’t already have one).
3. On the right-hand side of that box, look for the small **pencil icon (✏️)**.
4. Click the pencil.
5. A text field will appear where you can enter or edit:

   * **Description** → A short summary of what your project does.
   * **Website link (optional)** → A URL to your project page, documentation, or demo.
6. Once you’re done, press **Save** (or just hit Enter).

### Method 2: From repository settings

1. Go to your repository and click on the **Settings** tab (top menu bar).
2. Under **General**, scroll to the **Repository details** section.
3. There you’ll see fields for **Description** and **Website**.
4. Update them as needed.
5. Scroll down and click **Save changes** (if shown).

⚠️ **Note:** You’ll need **admin or maintainer access** to the repository in order to change its description. If you don’t see the pencil icon or the settings option, you may not have the right permissions.

After updating, the new description will appear at the top of your repo page and also be visible in GitHub search results and on your profile.

Hope this helps! 🚀

**Comment by sahalbinhussain-cloud** [comment]
https://github.com/orgs/community/discussions/54372#discussioncomment-6830666

**Comment by sahalbinhussain-cloud** [comment]
Short description of this repository
Website  What is the purpose of this
